### PR TITLE
Multi-arch support

### PR DESCRIPTION
**Pull request by detiber**

### CODE REVIEW

**Comment by thebsdbox** [comment]
Awesome work! @detiber 😀

**Comment by detiber** [comment]
@gianarb This is currently in a state where I've been able to validate that things are in a functional state:
- multi-arch kernel builds using buildx create functional kernels
- multi-arch container builds for tink-docker and bootkit using buildkit
- linuxkit builds for both amd64 and arm64 are functional.

In order for this to work for CI and automated deployment we'll likely need a few additional things:
- A self-hosted runner pre-configured with a buildx builder configured with native nodes for building the kernel image when needed. (Other use cases can likely remain on the default hosted runners)
- Container registry to push images to from the buildx builds, and appropriate credentials configured in the repo settings
- GitHub Workflow changes to support ^

Not sure how you want to proceed. I'm also wondering if it would make sense to migrate the project to the Tinkerbell org before trying to tackle the above.

**Comment by detiber** [comment]
Things look mostly functional now for the CI builds. The kernel ci build still has a minor issue (though not causing it to fail) related to pushing the multi-arch manifest (but not the underlying arch-specific images) to an insecure registry, caused by this buildx bug: https://github.com/docker/buildx/issues/354

If things look good, I can go ahead and rebase/squash the changes.

File: .github/workflows/kernel-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: For each Kernel PR
+on:
+  pull_request:
+    paths:
+    - kernel/**
```
Comment (thebsdbox) [inline .github/workflows/kernel-ci.yaml:5]: NICE, wondered how this hung together

**Review by thebsdbox — APPROVED** [review]
From my limited understanding of Makefiles, this all looks good to me 👍  Excellent stuff!

**Review by markyjackson-taulia — APPROVED** [review]
This is awesome work! 
/lgtm

**Comment by detiber** [comment]
Squashed, should be ready to go.

**Comment by gianarb** [comment]
Awesome @detiber !!! Thanks

**Pull request merged**